packages/dom/src/dom.js

```diff
@@ -453,6 +453,7 @@ export function isTextField( element ) {
 
 		return (
 			( nodeName === 'INPUT' && selectionStart !== null ) ||
+			element.getAttribute( 'type' ) === 'email' ||
```

Comment:
Probably should make this  `( nodeName === 'INPUT' && element.getAttribute( 'type' ) === 'email' )`  will update it if there is some agreement to go with this approach.

Reply:
I don't get it... is this input type not an input element?

Reply:
The problem is that Chrome at least does not set `selectStart` for anything other than `<input  type="text">`. For  `<input  type="email">`  `selectStart` is `null`, so this method returns false, but in all instances as far as I can tell we are going to want the block editor to treat an input of type email as a text input - sorry, I outlined this in the original ticket, but should have copied it over to this PR.

Reply:
https://stackoverflow.com/a/21959157 seems to be related and links to this [overview](https://html.spec.whatwg.org/multipage/input.html#do-not-apply). `type=number` should be affected as well.

Reply:
@ellatrix How would you like to see this fixed?

Reply:
Can we remove the `selectionStart` check? I don't know why it's there. There's no comment about it. In which commit was this added?

Reply:
This was added in https://github.com/WordPress/gutenberg/pull/5513/commits/31e70d58a519d9bfb21da5a9d29fb2fb29fa8cc1 by @aduth. Also changed in https://github.com/WordPress/gutenberg/commit/c05ede75a490b5c8b911ad43c6d882a5e61f0a22 to catch an exception in Safari related to `selectionStart`.

Reply:
Hm, I can't find the relevant PR for 31e70d5. Based on the comment in the other commit, it sounds like we should remove this check as it's terribly inconsistent.

Reply:
Sorry, the commit was part of #5513.

Reply:
Another idea might be restore the previous `isInputField()` and change `isTextField()` to use `isInputField()` + `selectionStart`. Then use `isInputField()` in `documentHasSelection()` which is the [method to check for the native copy handler](https://github.com/WordPress/gutenberg/blob/6b6af357f8addeb5754b620eb5d5e19f2775cf37/packages/block-editor/src/components/copy-handler/index.js#L39).

Reply:
@aduth Do have any thoughts on removing `selectionStart` from `isTextField()`?

Reply:
I could have been clearer with code comments. If I recall correctly, I think the main idea was to exclude certain `<input>` elements like "checkbox" and "button". If `selectionStart` is a problem for indicating the positive case of "is a text input", then maybe the logic can be inverted to the negative case of "is not an excluded input". The existing unit tests should give some coverage here.

Reply:
At a quick glance it seems that of the possible 17 `input` `types` 7 could be classed as `text` in terms of user using standard keyboard/paste for text input, so if we want to remove `selectionStart` it probably makes more sense to identify the positive cases explicitly, so have updated the PR to take this approach - if there is some agreement on this will also take a look at tests to check we have this covered effectively.